### PR TITLE
plotting progress

### PR DESCRIPTION
**Pull request by ozgunozerk**
@1devNdogs 
Leo I remember you already implemented how to use this new functionality and it was working (we were working on this branch). 
You can commit your changes too in this PR, and we can review it later together (and will also assign Nazar as a reviewer after we finalize this PR).

- [x] We also have to change the default 10GB allocated space with the actual data to be plotted, queried from the blockchain. (will do later in an upcoming PR)

### CODE REVIEW

File: src/pages/PlottingProgress.vue [hunk 1]
```diff
@@ -214,21 +217,62 @@ export default defineComponent({
...
+      
+      this.plottingData.status = lang.startingFarmer
+      // After local node is fully Synced, the farmer will be able to actualy plot and farm. 
+      const { publicKey, mnemonic } = await startFarming(this.plotDirectory.replace("/subspace.plot", ""))
+ 
+      // When farmer start, temporal mnemonic storage, when user accept on modal that mnemonic was backed up. We remove it from local storage
+      // Check SaveKeys.vue.
+      if (publicKey && mnemonic) {
+        // TODO: Fully remove LocalStorage. Find a secure way to pass mnemonic to SaveKeys modal.
+        LocalStorage.set("mnemonic", mnemonic)
```
Comment (ozgunozerk) [inline src/pages/PlottingProgress.vue:249]: I couldn't see how mnemonic is removed from LocalStorage after here? 
I saw a reset on mnemonic on here: https://github.com/subspace/subspace-desktop/pull/46/files#diff-653e0382a5f2a0bcb258b7abb55fe7ac7e6e00e6e9295618f526671a54d11cc5R66

But I believe that code will be executed before here?

File: src-tauri/src/main.rs [hunk 2]
```diff
@@ -151,8 +175,23 @@ pub(crate) async fn farm(
 
         move || Plot::open_or_create(&base_directory)
     });
+
     let plot = plot_fut.await.unwrap()?;
 
+    let mut _handler_guard = HANDLER
```
Comment (nazar-pc) [inline src-tauri/src/main.rs:181]: I don't understand why you have `let mut _handler_guard =` and why you need to store it at all. Maybe you can just call `.detach()` on handler instead?

File: src-tauri/src/main.rs [hunk 3]
```diff
@@ -6,11 +6,17 @@
 #[cfg(target_os = "windows")]
 mod windows;
 
+#[macro_use]
+extern crate lazy_static;
```
Comment (nazar-pc) [inline src-tauri/src/main.rs:10]: `extern crate` is 2015 Rust style, use `use` instead.

**Review by 1devNdogs — APPROVED** [review]

File: src-tauri/src/main.rs (hunk 2 above)
Comment (nazar-pc) [reply src-tauri/src/main.rs]: Why do you need to store handler at all?

Comment (ozgunozerk) [reply src-tauri/src/main.rs]: I don't. It was there because of the old trials. I don't store it, this is changed with the last commit

Comment (ozgunozerk) [reply src-tauri/src/main.rs]: I thought you were referring to `let mut _handler_guard`. 
So I need a `HANDLER` to be able to keep track of updates on the plotting pieces. 
This needs to be initiated in `farm` function. However, `farm` function returns quickly, and because of that the `HANDLER` needs to be a global variable to outlive the farmer. At least, that was the solution I came up with.

Comment (nazar-pc) [reply src-tauri/src/main.rs]: I was referring both to that (because it wasn't even a guard) and the fact that you store anything in the first place. I also mentioned method that you can call instead of storing it and achieve the same result with less dependencies and less code.

Comment (ozgunozerk) [reply src-tauri/src/main.rs]: I think it is fine now :)

**Review by nazar-pc — APPROVED** [review]
